CMakeLists.txt

```diff
@@ -87,6 +87,11 @@ elseif(CMAKE_C_COMPILER_ID STREQUAL "Intel")
 endif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
 
 #-----------------------------------------------------------------------
+# Check for prerequisite libraries
+
+find_package(Threads)
```

Comment:
No conflict; I've just rearranged the file so that all of the option definitions appear before any of the prereq checks.